File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlForXmlRootDirective.cs
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlForXmlRootDirective : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.ForXmlRootDirective;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ForXmlRootDirective(this);
    }

    public ISqlExpression? RootName { get; set; }

    public string ToSql()
    {
        var sql = new IndentStringBuilder();
        sql.Write("ROOT");
        if (RootName != null)
        {
            sql.Write("(");
            sql.Write(RootName.ToSql());
            sql.Write(")");
        }
        return sql.ToString();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlForXmlRootDirective.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlForXmlRootDirective implements ISqlExpression {
    SqlType: SqlType = SqlType.ForXmlRootDirective;
    Span: TextSpan = new TextSpan();
    RootName?: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_ForXmlRootDirective(this);
    }

    ToSql(): string {
        return 'FOR XML ROOT';
    }
} 